packages/kit/src/vite/index.js

```diff
@@ -61,7 +62,53 @@ const enforced_config = {
  * @return {import('vite').Plugin[]}
  */
 export function sveltekit() {
-	return [...svelte(), kit()];
+	return [...svelte(), env(), kit()];
+}
+
+/**
+ * Returns the SvelteKit environment plugin. Replaces `$env/static/*` imports
+ * with the actual environment variables.
+ * @return {import('vite').Plugin}
+ */
+function env() {
+	/** @type {import('types').ValidatedConfig} */
+	let svelte_config;
+
+	/** @type {import('vite').ConfigEnv} */
+	let vite_config_env;
+
+	/** @typedef {{public: {[k: string]: string;};private: {[k: string]: string;};}} ResolvedEnv */
+
+	/** @type {ResolvedEnv} */
+	let resolved_environment;
+
+	return {
+		name: 'vite-plugin-svelte-kit:env',
+
+		async config(_, config_env) {
+			vite_config_env = config_env;
+			svelte_config = await load_config();
```

Comment:
there were previous efforts to avoid reading svelte config multiple times, as it may lead to unexpected side effects when it contains TLA initialization code.

To avoid this here we'd either have to steal the ref from kit plugin somehow or move the resolve + load hooks there

cc @benmccann

Reply:
I thought we could load multiple times as long as we use the same loading helper (just have to make sure the loading helper uses the mtime trick)

Reply:
I'll let you guys duke this one out as I have no idea one way or the other. I like this as its own plugin (it's just way less confusing and way more obvious what it's doing), but I can move it back into the main plugin if there's a need.

Reply:
I moved it into the main plugin — if we do get round to creating a helper that allows us to load the config multiple times without penalty then we should, but in the meantime i think it's better that we load it as few times as possible